File: VB/LookUpEditFilterExample/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System
Imports System.Collections.ObjectModel

Namespace LookUpEditFilterExample.ViewModels
    <POCOViewModel> _
    Public Class MainViewModel
        Protected Sub New()
            Data = New ObservableCollection(Of Data)()
            Dim r As New Random()
            For i As Integer = 0 To 24
                Data.Add(New Data() With { _
                    .ID = i, _
                    .Name = "Name " & i, _
                    .Value = r.Next(0, 25) _
                })
            Next i
        End Sub

        Public Overridable Property Data() As ObservableCollection(Of Data)
    End Class

    Public Class Data
        Public Property ID() As Integer
        Public Property Name() As String
        Public Property Value() As Integer
    End Class
End Namespace

File: CS/LookUpEditFilterExample/ViewModels/MainViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Collections.ObjectModel;

namespace LookUpEditFilterExample.ViewModels {
    [POCOViewModel]
    public class MainViewModel {
        protected MainViewModel() {
            Data = new ObservableCollection<Data>();
            Random r = new Random();
            for (int i = 0; i < 25; i++)
                Data.Add(new Data() { ID = i, Name = "Name " + i, Value = r.Next(0, 25) });
        }

        public virtual ObservableCollection<Data> Data { get; set; }
    }

    public class Data {
        public int ID { get; set; }
        public string Name { get; set; }
        public int Value { get; set; }
    }
}